src/basic/fs-util.c

```diff
@@ -742,17 +742,23 @@ bool unsafe_transition(const struct stat *a, const struct stat *b) {
 }
 
 static int log_unsafe_transition(int a, int b, const char *path, unsigned flags) {
-        _cleanup_free_ char *n1 = NULL, *n2 = NULL;
+        _cleanup_free_ char *n1 = NULL, *n2 = NULL, *user_a = NULL, *user_b = NULL;
+        struct stat stat_a, stat_b;
 
         if (!FLAGS_SET(flags, CHASE_WARN))
                 return -ENOLINK;
 
         (void) fd_get_path(a, &n1);
         (void) fd_get_path(b, &n2);
 
+        if (fstat(a, &stat_a) == 0)
+            user_a = uid_to_name(stat_a.st_uid);
+        if (fstat(b, &stat_b) == 0)
+            user_b = uid_to_name(stat_b.st_uid);
```

Comment:
just one more small nit: you can use a single `struct stat st` variable, no need to have two of them.

Reply:
Good idea